aws/types.go

```diff
@@ -66,6 +66,14 @@ func Double(v float64) *float64 {
 	return &v
 }
 
+//A TimeValue is a time.Time object which may or may not be present.
+type TimeValue *time.Time
+
+//Time converts a time.Time into a TimeValue
+func Time(t time.Time) *time.Time {
```

Comment:
I think that's much more elegant - I originally thought to do that but saw the other types in `types.go` not using the type alias they had just declared as the return type. Is there any reason why the other types are written that way?